src/plugins/badge.coffee

```diff
@@ -0,0 +1,583 @@
+  </div>
+  """
+
+  STAR_TEMPLATE = (type, assigns) -> """
+  <div class="sa-badge-star sa-badge-#{type}-star"></div>
+  """
+
+  MODAL_INNER_TEMPLATE = (assigns) -> """
+  <div id="sa-badge-modal-inner" class="sa-badge-spinner">
+    <span id="sa-badge-modal-close-button"></span>
+      <iframe id="sa-badge-modal-iframe" src="#{assigns.src}" width="100%" height="100%" frameborder="0"></iframe>
+    </div>
+  </div>
+  """
+
+  constructor: ->
+    @_setParentDoc()
+
+    @addStyle()
+    @onDOMReady => @initialize()
+
+  initialize: ->
+    return unless @_canRender()
+
+    @render()
+    @_bindHandlers()
+
+  ###
+  Detect DOM ready
+  @todo DRY this method as it is being used in order_stash plugin too.
+  ###
+  onDOMReady: (cb) ->
+    # IE is only safe when readyState is 'complete'
+    return cb() if @parent_doc.readyState == 'complete'
+    return cb() if !@parent_doc.attachEvent && @parent_doc.readyState in ['interactive', 'complete']
+
+    if @parent_doc.addEventListener # Mozilla, Opera, Webkit
+      @parent_doc.addEventListener 'DOMContentLoaded', cb, false
+    else if @parent_doc.attachEvent # IE
+      @parent_doc.attachEvent('onreadystatechange', (=> cb() if @parent_doc.readyState == 'complete'))
+    else
+      setTimeout((=> try cb() catch), DOMREADY_TIMEOUT)
+
+  ###
+  Displays the plugin
+  ###
+  render: ->
+    switch context().configuration.display
+      when 'floating' then @_renderFloating()
+      when 'embedded' then @_renderEmbedded()
+
+  $floatingBadgePlugin: -> @parent_doc.getElementById('sa-badge-floating-plugin')
+  $embeddedBadgePlugin: -> @parent_doc.getElementById('sa-badge-embedded-plugin')
+  $embeddedBadgeMoreButton: -> @parent_doc.getElementById('sa-badge-embedded-more-button')
+  $closeModalButton: -> @parent_doc.getElementById('sa-badge-modal-close-button')
+  $showReviewsButton: ->
+    switch context().configuration.display
+      when 'floating' then @$floatingBadgePlugin()
+      when 'embedded' then @$embeddedBadgeMoreButton()
+
+  ###
+  Adds the required css for the plugin
+  ###
+  addStyle: ->
+    style = document.createElement('style')
+    style.id = 'sa-badge-style'
+    style.type = 'text/css'
+
+    # IE compatibility
+    if style.styleSheet
+      style.styleSheet.cssText = STYLE(context().configuration)
+    else
+      style.appendChild(document.createTextNode(STYLE(context().configuration)))
+
+    @_head().appendChild(style)
+
+  _renderFloating: ->
+    $el = document.createElement('div')
+    $el.id = 'sa-badge-floating-plugin'
+    $el.innerHTML = FLOATING_TEMPLATE(stars: @_ratingToStars(context().data.rating))
+    @parent_doc.body.appendChild($el)
+
+  _renderEmbedded: ->
+    rating = context().data.rating
+    @$embeddedBadgePlugin().innerHTML = EMBEDDED_TEMPLATE(rating: rating.toFixed(1), stars: @_ratingToStars(rating))
+
+  _bindHandlers: ->
+    @_attachEvent(@$showReviewsButton(), 'click', @_onClickShowReviewsButton)
+
+  _attachEvent: (element, event, handler) ->
```

Comment:
Add `@todo DRY this method as it is being used in order_stash plugin too.` as in `addStyle`